crates/debugging/src/trace.rs

```diff
@@ -0,0 +1,81 @@
+use crate::tree::node::Node;
+use crate::tree::serialize::TreeSerialize;
+use blockifier::execution::entry_point::CallType;
+use cheatnet::runtime_extensions::call_to_blockifier_runtime_extension::rpc::CallResult;
+use cheatnet::state::{CallTrace, CallTraceNode};
+use starknet_api::contract_class::EntryPointType;
+use starknet_api::core::{ContractAddress, EntryPointSelector};
+use starknet_api::transaction::fields::Calldata;
+use std::cell::RefCell;
+use std::fmt::Display;
+use std::rc::Rc;
+
+#[derive(Debug, Clone)]
+pub struct Trace {
+    pub selector: EntryPointSelector,
+    pub trace_info: TraceInfo,
+}
+
+#[derive(Debug, Clone)]
+pub struct TraceInfo {
+    pub entry_point_type: EntryPointType,
+    pub calldata: Calldata,
+    pub storage_address: StorageAddress,
+    pub caller_address: CallerAddress,
+    pub call_type: CallType,
+    pub nested_calls: Vec<Trace>,
+    pub call_result: CallResult,
+}
+
+/// Wrapper for [`ContractAddress`] to distinguish storage addresses at the type level.
+#[derive(Debug, Clone)]
+pub struct StorageAddress {
+    pub address: ContractAddress,
+}
```

Comment:
Nit but maybe let's just use this if it's just a wrapper. Don't have a strong opinion on it though.

```suggestion
pub struct StorageAddress(ContractAddress)
```

Reply:
To be honest, I don't like tuple structs. Accessing fields via `.0` feels weird to me. I would prefer using this